lesson_03/script.js

```diff
@@ -0,0 +1,16 @@
+"use strict";
+const ul = document.getElementById("js-target");
+const data = ["a1.html", "a2.html"];
+const listCount = function () {
+  return 2;
+};
+for (let i = 1; i <= listCount(); i += 1) {
+  const li = document.createElement("li");
+  const a = document.createElement("a");
+  const img = document.createElement("img");
+  a.text = `a${i}`;
+  a.href = `${data[i - 1]}`;
+  img.src = "/img/bookmark.png";
+
+  ul.appendChild(li).appendChild(a).insertAdjacentElement("afterbegin", img);
```

Comment:
Please try using `document.createDocumentFragment`.
If you use appendChild in a loop, the performance will be bad.
It's better to use and add that fragment at the end.

```
const fragment = document.createDocumentFragment();
{ // ループ内
fragment.appendChild(li).appendChild(a).insertBefore(img, a.firstChild);  
}
//最後にadd
ul.appendChild(fragment)

```